docs/advanced.rst

```diff
@@ -508,6 +508,143 @@ Beside some standard information, like the *boot ID*, the system's *uptime* or
+without automatic installation.
+
+Automatic Installation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The criteria which a bundle needs to fulfill to be considered a valid update can
+be configured in via ``candidate-criteria`` in the ``[poll]`` section.
+For any valid candidate, the ``install-criteria`` option can be used to trigger
+an automatic installation (see the :ref:`reference <poll-candidate-criteria>`
+for supported criteria).
+If no automatic installation is triggered, it can be confirmed explicitly (see
+below).
+
+After an automatic installation completes, the ``reboot-criteria`` are checked
+to determine if the system should be rebooted.
+If the installation of a bundle fails, it is not attempted again until the RAUC
+service is restarted or a new manifest is found.
+
+Example usage:
+
+.. code-block:: cfg
+   :emphasize-lines: 3-5
+
+   [poll]
+   source=https://example.com/stable/my_product.raucb
+   candidate-criteria=different-version
+   install-criteria=higher-semver
+   reboot-criteria=updated-slots;updated-artifacts
+
+In this example:
+
+1. A new bundle is considered a valid update candidate if it's declared version
+   is different from the current system version.
+2. It is automatically installed only if its semantic version is higher.
+3. A reboot occurs automatically after installation **only** if a slot or
+   artifact was actually installed.
+
+For valid candidates which don't have a higher version, the update is only
+offered for confirmation.
+
+Confirmed Installation
+~~~~~~~~~~~~~~~~~~~~~~
+
+For more complex cases, RAUC can handle the polling and inform a separate
+application or the user if an update is available.
+If and when an update should be installed, the :ref:`InstallBundle D-Bus method
+<gdbus-method-de-pengutronix-rauc-Installer.InstallBundle>` is used to trigger
+the actual installation (see below).
+
+To ensure that the correct bundle is installed after evaluating the manifest
+information, RAUC uses the "manifest hash".
+This hash is computed over the signed manifest and fully identifies the contents
+of the bundle.
+Note that the signature itself is *not* included in the manifest hash.
+
+Backoff
+~~~~~~~
+
+If RAUC is unable to fetch a valid bundle manifest from the URL, it will
+increase the polling interval to avoid an overload scenario.
+Each consecutive failure extends the base interval (``interval-sec``) by the
+length of the base interval.
+The interval is not extended beyond the maximum set by ``max-interval-sec``,
+which defaults to four times the base interval.
+
+Example usage:
+
+.. code-block:: cfg
+   :emphasize-lines: 3-4
+
+   [poll]
+   source=https://example.com/stable/my_product.raucb
+   interval-sec=300
+   max-interval-sec=3600
+
+In this example:
+
+* RAUC polls the source every 5 minutes as long as no error occurs.
+* Every time a poll attempt fails, the interval is increased by 5 minutes, until
+  it reaches 1 hour.
+* If an attempt is successful again, the interval is reduced to 5 minutes.
```

Comment:
The documentation should contain best practices. Polling every five minutes sounds excessive to me.

How about e.g.

```
   interval-sec=14400
   max-interval-sec=86400
```

e.g. four hours and one day respectively. Four hours still sounds a bit too frequent for my taste, but at least [hacker news will not make fun of it](https://news.ycombinator.com/item?id=43829006).